addons/mail/controllers/discuss.py

```diff
@@ -406,16 +406,30 @@ def mail_channel_add_guest_as_member(self, channel_id, channel_uuid, **kwargs):
                 raise NotFound()
 
     @http.route('/mail/channel/messages', methods=['POST'], type='json', auth='public')
-    def mail_channel_messages(self, channel_id, max_id=None, min_id=None, limit=30, **kwargs):
+    def mail_channel_messages(self, channel_id, max_id=None, min_id=None, limit=30, order=None, **kwargs):
         channel_member_sudo = request.env['mail.channel.member']._get_as_sudo_from_request_or_raise(request=request, channel_id=int(channel_id))
-        messages = channel_member_sudo.env['mail.message']._message_fetch(domain=[
+        domain = [
             ('res_id', '=', channel_id),
             ('model', '=', 'mail.channel'),
             ('message_type', '!=', 'user_notification'),
-        ], max_id=max_id, min_id=min_id, limit=limit)
-        if not request.env.user._is_public():
+        ]
+        if 'around' in kwargs:
+            messages_before = channel_member_sudo.env['mail.message']._message_fetch(domain=[*domain, ('id', '<=', kwargs['around'])], limit=limit // 2)
+            messages_after = channel_member_sudo.env['mail.message']._message_fetch(domain=[*domain, ('id', '>', kwargs['around'])], limit=limit // 2, order='id')
```

Comment:
`order` here would be more readable if one was specific "id asc", and the other specific "id desc", if I'm understanding correctly the goal.